models/orogen/rock_gazebo.rb

```diff
@@ -1,7 +1,12 @@
+# frozen_string_literal: true
+
 require 'common_models/models/services/joints_control_loop'
 require 'common_models/models/services/transformation'
 require 'common_models/models/devices/gazebo'
 
+## Double negation is needed to convert objects to boolean when writing to properties
```

Comment:
Not really needed... There are several other ways to do this other than using double negation.

That said, `!!` is a pretty neat idiom and if that's what you prefer to have in the codebase maybe we should just disable this cop altogether. Same applies for the ones regarding complexity: if we are disabling and/or ignoring them everywhere why not disable in the gem once and for all?

Reply:
> Same applies for the ones regarding complexity: if we are disabling and/or ignoring them everywhere why not disable in the gem once and for all?

I agree we have to weed out a few things. Ideally, I'd like to disable the block length cop only on test files. Complexity, I'm **not** ignoring. I'm not fixing them everywhere systematically, but follow them on new code.

For the double-negation, it actually makes sense only in a few places (where we have interface with e.g. JSON or Typelib). So, to me, it makes sense to keep it.

Reply:
> it actually makes sense only in a few places

I don't agree. But, I am fine with it anyway